pkg/operator/ceph/object/objectstore.go

```diff
@@ -813,13 +813,10 @@ func ConfigureSharedPoolsForZone(objContext *Context, sharedPools cephv1.ObjectS
 	// persist configuration updates:
 	if hasZoneChanged {
 		logger.Infof("zone config changed: performing zone config updates for %s", objContext.Zone)
-		updatedZoneResult, err := updateZoneJSON(objContext, zoneUpdated)
+		_, err := updateZoneJSON(objContext, zoneUpdated)
 		if err != nil {
 			return fmt.Errorf("unable to persist zone config update for %s: %w", objContext.Zone, err)
 		}
-		if err = zoneUpdateWorkaround(objContext, zoneUpdated, updatedZoneResult); err != nil {
-			return fmt.Errorf("failed to apply zone set workaround: %w", err)
-		}
 	}
```

Comment:
I think @arttor added the workaround code. It might be good to have him double check this just to make sure we don't anticpiate any issues.